app/assets/javascripts/discourse/components/topic-list-item.js.es6

```diff
@@ -64,6 +75,16 @@ export const ListItemDefaults = {
     }
   },
 
+  @observes("bulkSelectEnabled", "topic.{pinned,tags}")
+  rerenderTriggers() {
+    this.renderTopicListItem();
+  },
+
+  @discourseComputed("topic.{archived,visible,closed,category}")
```

Comment:
hmmm this weird

computedProperties are supposed to a value, here you return nothing, so nothing ever get cached, so i'm not even sure what it does 😅

Reply:
It has to do with the bulk selected items and the actions you take on them. If you bulk select a bunch of topics, then use the wrench to close them all, this will trigger a re-render of the raw template and show the closed icon next to each topic. Without this the icons never get added unless you hard-refresh the page.

![image](https://user-images.githubusercontent.com/1490496/71136104-e9501b00-21c0-11ea-8073-89ccf6b82620.png)

Reply:
I'm curious if only having:

```
didReceiveAttrs() {
  this._super(...arguments);
  this.renderTopicListItem();
}
```

Work? (You can probably remove init with this too.)

Reply:
Okay updated it to use didReceiveAttrs which is working great! Thanks!

I'm not sure why I didn't use it this time cause that was the pattern I've been using for all the other updates I've done.

The only thing which still needs an observer now is just for pinning a topic, because you can actually click the pin and toggle it. None of the other icons have this functionality.